pkg/subctl/cmd/validate_all.go

```diff
@@ -52,22 +56,40 @@ func validateAll(cmd *cobra.Command, args []string) {
 		status.End(cli.Success)
 		fmt.Println()
 
-		validateCNIInCluster(item.config, item.clusterName, submariner)
+		if !validateCNIInCluster(item.config, item.clusterName, submariner) {
+			validationStatus = false
+		}
 		fmt.Println()
-		validateConnectionsInCluster(item.config, item.clusterName)
+		if !validateConnectionsInCluster(item.config, item.clusterName) {
+			validationStatus = false
+		}
 		fmt.Println()
-		checkPods(item, submariner, OperatorNamespace)
+		if !checkPods(item, submariner, OperatorNamespace) {
+			validationStatus = false
+		}
 		fmt.Println()
-		checkOverlappingCIDRs(item, submariner)
+		if !checkOverlappingCIDRs(item, submariner) {
+			validationStatus = false
+		}
 		fmt.Println()
-		validateKubeProxyModeInCluster(item.config, item.clusterName)
+		if !validateKubeProxyModeInCluster(item.config, item.clusterName) {
+			validationStatus = false
+		}
 		fmt.Println()
-		validateFirewallMetricsConfigWithinCluster(item.config, item.clusterName)
+		if !validateFirewallMetricsConfigWithinCluster(item.config, item.clusterName) {
+			validationStatus = false
+		}
 		fmt.Println()
-		validateVxLANConfigWithinCluster(item.config, item.clusterName, submariner)
+		if !validateVxLANConfigWithinCluster(item.config, item.clusterName, submariner) {
+			validationStatus = false
+		}
 		fmt.Println()
 		fmt.Printf("Skipping tunnel firewall validation as it requires two kubeconfigs." +
 			" Please run \"subctl validate firewall tunnel\" command manually.\n")
 		fmt.Println()
 	}
+
+	if !validationStatus {
+		exitWithErrorMsg("Diagnose all return with an error")
```

Comment:
Any errors that occurred earlier were already outputted so I don't think we need an extra message here - we can just exit. Same elsewhere.

```suggestion
		os.Exit(1)
```